src/xtgeoapp_grd3dmaps/aggregate/_co2_mass.py

```diff
@@ -452,114 +471,122 @@ def generate_co2_mass_data(
-        mass_gas_phase_prop = xtgeo.grid3d.GridProperty(ncol=grid_pf.ncol,nrow=grid_pf.nrow,nlay=grid_pf.nlay,values=mass_gas_phase_array,name=mass_gas_phase_name,date=str(x.date))
-
-        if maps is None:
-            maps = []
-        elif isinstance(maps,str):
-            maps = [maps]
-        maps = [map_name.lower() for map_name in maps]
-        
-        if "all" in maps or len(maps)==0:
-            mass_total_prop.to_file(grid_out_dir + "/MASS_TOTAL_"+str(x.date)+".roff", fformat="roff")
-            mass_aqu_phase_prop.to_file(grid_out_dir + "/MASS_AQU_PHASE_"+str(x.date)+".roff", fformat="roff")
-            mass_gas_phase_prop.to_file(grid_out_dir + "/MASS_GAS_PHASE_"+str(x.date)+".roff", fformat="roff")
-            mass_total_prop_list.append(mass_total_prop)
-            mass_aqu_phase_prop_list.append(mass_aqu_phase_prop)
-            mass_gas_phase_prop_list.append(mass_gas_phase_prop)
-            all_maps_bool = True
-        if "free_co2" in maps and all_maps_bool==False:
-            mass_gas_phase_prop.to_file(grid_out_dir + "/MASS_GAS_PHASE_"+str(x.date)+".roff", fformat="roff")
-            mass_gas_phase_prop_list.append(mass_gas_phase_prop)
-        if "dissolved_co2" in maps and all_maps_bool==False:
-            mass_aqu_phase_prop.to_file(grid_out_dir + "/MASS_AQU_PHASE_"+str(x.date)+".roff", fformat="roff")
-            mass_aqu_phase_prop_list.append(mass_aqu_phase_prop)
-        if "total_co2" in maps and all_maps_bool==False:
-            mass_total_prop.to_file(grid_out_dir + "/MASS_TOTAL_"+str(x.date)+".roff", fformat="roff")
-            mass_total_prop_list.append(mass_total_prop)
-
-    out_list = [mass_gas_phase_prop_list,mass_aqu_phase_prop_list,mass_total_prop_list]
-    return out_list
-
-def _temp_make_property_copy(source: str, grid_file: Optional[str], dates: List[str]) -> xtgeo.GridProperty:
+    triplets = [
+        (int(row["IX"] - 1), int(row["JY"] - 1), int(row["KZ"] - 1))
+        for _, row in gdf.iterrows()
+    ]
+    return dimensions, triplets
+
+
+def _convert_to_grid(
+    co2_at_date: Co2DataAtTimeStep,
+    dimensions: Tuple[int, int, int],
+    triplets: List[Tuple[int, int, int]],
+) -> Dict[str, xtgeo.GridProperty]:
+    """
+    Store the CO2 mass arrays in grid objects
+    """
+    grids = {}
+    date = str(co2_at_date.date)
+    for mass, name in zip(
+        [co2_at_date.total_mass(), co2_at_date.aqu_phase, co2_at_date.gas_phase],
+        ["mass-total", "mass-aqu-phase", "mass-gas-phase"],
+    ):
+        mass_array = np.zeros(dimensions)
+        for i, triplet in enumerate(triplets):
+            mass_array[triplet] = mass[i]
+        mass_name = "co2-" + name + "--" + date
+        grids[name] = xtgeo.grid3d.GridProperty(
+            ncol=dimensions[0],
+            nrow=dimensions[1],
+            nlay=dimensions[2],
+            values=mass_array,
+            name=mass_name,
+            date=date,
+        )
+    return grids
+
+
+def _temp_make_property_copy(
```

Comment:
Is this method used? If so we can change the name, just remove "temp". The name was made when it was not clear if this was a method that was going to be used.

Reply:
Removed now